### ISSUE TITLE
Step Functions JSONata isn't fully supported - "Arguments" and "QueryLanguage" properties are not allowed

### ISSUE THREAD

**Issue by tamir-deep** [aws-cloudformation/cfn-lint#3907]
### CloudFormation Lint Version

1.22.4

### What operating system are you using?

MacOS

### Describe the bug

AWS [not long ago](https://aws.amazon.com/blogs/compute/simplifying-developer-experience-with-variables-and-jsonata-in-aws-step-functions/) introduced JSONata support in Step Functions.

CFN-Lint keeps throwing errors when using JSONata within the SF state machine's definition properly:
```
E3601     Additional properties are not allowed ('Arguments' was unexpected)
E3601     Additional properties are not allowed ('QueryLanguage' was unexpected)
```

The template below does work in CloudFormation itself of course.

Relates also to Issue #3858

### Expected behavior

CFN-Lint will not detect JSONata syntax in SF as errors.

### Reproduction template

```yaml
Resources:
  LogGroup:
    Type: AWS::Logs::LogGroup
    UpdateReplacePolicy: Delete
    DeletionPolicy: Delete
    Properties:
      RetentionInDays: 14

  StateMachinee5ae0411:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      Definition:
        StartAt: CreateQueue
        States:
          CreateQueue:
            Next: DeleteQueue
            Resource: arn:aws:states:::aws-sdk:sqs:createQueue
            Type: Task
            Assign:
              inputPayload: '{% $states.context.Execution.Input %}'
              QueueUrl: '{% $states.result.QueueUrl %}'
            QueryLanguage: JSONata
            Arguments:
              Attributes:
                MessageRetentionPeriod: '345600'
                ReceiveMessageWaitTimeSeconds: '20'
                VisibilityTimeout: '120'
              QueueName: '{% ''prefixxxxxx-'' & $states.input.bucketName %}'
              Tags:
                Bucket: '{% $states.input.bucketName %}'
          DeleteQueue:
            Type: Task
            Resource: arn:aws:states:::aws-sdk:sqs:deleteQueue
            Next: Success
            QueryLanguage: JSONata
            Arguments:
              QueueUrl: '{% $QueueUrl %}'
          Success:
            Type: Succeed
      StateMachineName: StateMachinee5ae0411
      StateMachineType: STANDARD
      RoleArn: "arn:aws:blababla"
      EncryptionConfiguration:
        Type: AWS_OWNED_KEY
      LoggingConfiguration:
        Level: ERROR
        IncludeExecutionData: true
        Destinations:
          - CloudWatchLogsLogGroup:
              LogGroupArn: !GetAtt LogGroup.Arn
```

**Comment by kddejong** [comment]
Fix is merged into main and I'll try to release today.
